crates/diesel_models/src/authorization.rs

```diff
@@ -57,6 +57,20 @@ pub struct AuthorizationUpdateInternal {
     pub connector_authorization_id: Option<String>,
 }
 
+impl AuthorizationUpdateInternal {
+    pub fn create_authorization(self, source: Authorization) -> Authorization {
+        Authorization {
+            status: self.status.unwrap_or(source.status),
+            error_code: self.error_code,
```

Comment:
If error_code or error_message is None, instead of overriding the original values, can we use the source instead?